### ISSUE TITLE
Local Bucket:  tryConsumeAsMuchAsPossible returns negative values (and violates rate limit)

### ISSUE THREAD

**Issue by simon-baatz** [bucket4j/bucket4j#417]
I am using a Bucket (build using `Bucket.builder()`) from bucket4j 8.5.0. It is build like this:

```
Bucket.builder().addLimit( Bandwidth.builder().capacity( 50 ).refillGreedy( 50, Duration.ofSeconds( 1 ) ).build() ).build();
```

Since there are operations which I can't control but which need to count against the limit, `consumeIgnoringRateLimits` is used, probably resulting in a (sometimes) negative number of tokens.

When calling `tryConsumeAsMuchAsPossible` (with a positive limit), two things happen:

1. It may return a negative consumed token count (contrary to how I read its documentation)
2. It looks like this amount of negative tokens is actually "consumed", i.e. effectively _adding_ tokens to the bucket.

**Comment by vladimir-bukhtoyarov** [comment]
Bug there https://github.com/bucket4j/bucket4j/blob/67d688d18dd238a9838ea6a37e451404c38fa189/bucket4j-core/src/main/java/io/github/bucket4j/local/LockFreeBucket.java#L71 it needs to check <=0

I will provide fix at monday.

**Comment by simon-baatz** [comment]
Great! I can give it a spin once it's ready if you like.

**Comment by vladimir-bukhtoyarov** [comment]
Fix has been deployed with version ```8.6.0```.

**Comment by simon-baatz** [comment]
> Fix has been deployed with version `8.6.0`.

I upgraded to the 8.6.0 package and I can confirm that it fixes the problem I have seen with 8.5.0. Thank you for the quick resolution!
